packages/web/src/components/admin_page/feature_switch_page/render_switch.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { Button, Form, OverlayTrigger, Popover, Alert } from 'react-bootstrap';
+import axios from 'axios';
+import { parseToNewAPI, featureSwitchesURL, HEADERS } from '../../../Global';
+import FeatureRow from './row_switch';
+import { FsContext } from '../../../feature_switches';
+
+const UpdateSwitch = ({ onUpdate }) => {
+    const [newSwitch, setNewSwitch] = useState({
+        switch_id: '', // Initialize with an empty string
+        is_enabled: true, // Default value for the dropdown as a boolean
+    });
+
+    const handleChange = e => {
+        const { name, value } = e.target;
+        setNewSwitch(prevState => ({
+            ...prevState,
+            [name]: name === 'is_enabled' ? value === 'true' : value,
+        }));
+    };
+
+    const handleSubmit = e => {
+        e.preventDefault();
+        if (newSwitch.switch_id.trim() !== '') {
+            onUpdate(newSwitch);
+        }
+    };
+
+    return (
+        <Form onSubmit={handleSubmit}>
+            <Form.Group className="mb-3" controlId="formBasicEmail">
+                <Form.Label>Set a feature:</Form.Label>
+                <Form.Control
+                    type="text"
+                    placeholder="Enter new feature switch"
+                    onChange={handleChange}
+                    name="switch_id"
+                    value={newSwitch.switch_id}
+                    required
+                />
+                <Form.Text className="text-muted">Add a new feature</Form.Text>
+            </Form.Group>
+            <Form.Group className="mb-3" controlId="formBasicIsEnabled">
+                <Form.Label>Set it to true or false:</Form.Label>
+                <Form.Select onChange={handleChange} name="is_enabled" value={newSwitch.is_enabled.toString()}>
+                    <option value="true">True</option>
+                    <option value="false">False</option>
+                </Form.Select>
+                <Form.Text className="text-muted"></Form.Text>
+            </Form.Group>
+
+            <Button type="submit" variant="primary">
+                Add
+            </Button>
+        </Form>
+    );
+};
+
+export const popover = (onUpdate, errorMessage) => {
+    return (
+        <Popover id="popover-basic">
+            <Popover.Header as="h3">Want to update or remove?</Popover.Header>
+            <Popover.Body>
+                {errorMessage !== undefined ? <Alert variant="warning">{errorMessage}</Alert> : null}
+                <UpdateSwitch onUpdate={onUpdate} />
+            </Popover.Body>
+        </Popover>
+    );
+};
+
+const updateSwitches = (switches, setSwitches) => data => {
+    setSwitches({ ...switches, [data.switch_id]: data.is_enabled });
```

Comment:
The idea was to leave comments on top of such invocations. Briefly explain how and why you spread like that.